qiskit_metal/_gui/widgets/qlibrary_display/delegate_qlibrary.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Delegate for display of QComponents in Library tab
+"""
+
+from PySide2.QtWidgets import QItemDelegate
+from PySide2.QtCore import QAbstractProxyModel, QModelIndex, Qt
+import PySide2
+from PySide2.QtGui import QTextDocument
+from qiskit_metal._gui.widgets.qlibrary_display.file_model_qlibrary import QFileSystemLibraryModel
+
+USER_COMP_DIR = "user_components"
+
+
+class LibraryDelegate(QItemDelegate):
+    """
+    Delegate for QLibrary view
+    Requires LibraryModel
+    """
+
+    def __init__(self, parent=None):
+        """
+        source_model_type - The Delegate may belong to a view using a ProxyModel.
+        However, the source model for that Proxy Model(s) should be a QFileSystemLibraryModel
+        is_dev_mode - Whether the MetalGUI is in Developer Mode or not
+
+        Args:
+            parent: parent
```

Comment:
Throughout this file"
- Update docstring to include the return type 
- Update args to include argument types
- Update docstring to include exceptions